pkg/goversion/compat.go

```diff
@@ -0,0 +1,28 @@
+package goversion
+
+import (
+	"fmt"
+)
+
+var (
+	minSupportedVersionOfGoMinor = 10
+	maxSupportedVersionOfGoMinor = 12
+	goTooOldErr                  = fmt.Errorf("Version of Go is too old for this version of Delve (minimum supported version 1.%d)", minSupportedVersionOfGoMinor)
+	dlvTooOldErr                 = fmt.Errorf("Version of Delve is too old for this version of Go (maximum supported version 1.%d)", maxSupportedVersionOfGoMinor)
+)
+
+// Compatible checks that the currently installed version is compatible with
+// this version of delve.
+func Compatible() error {
+	ver, ok := Installed()
+	if ver.IsDevel() || !ok {
+		return nil
+	}
+	if !ver.AfterOrEqual(GoVersion{1, minSupportedVersionOfGoMinor, -1, 0, 0, ""}) {
+		return goTooOldErr
+	}
+	if ver.AfterOrEqual(GoVersion{1, maxSupportedVersionOfGoMinor + 1, -1, 0, 0, ""}) {
```

Comment:
Wouldn't this not catch issues like the Go 1.11 one?
E.g. if a breaking change happens in Go 1.13 then this check would still happily pass, when instead it should fail because we know nothing about it at the time of compilation for this version of delve.
Also, would this change check against the Go version that compiled the target process than the Go version that compiled delve?

Reply:
The check only passes for 1.10 through 1.12 which we know we support at this time. 1.13 would be rejected, until we make a version of delve that specifically supports it.

Reply:
🤦‍♂️ sorry, misread the condition.
Finally, does it take the go version of the compiled binary or of the system where delve is running?

Reply:
it takes the version returned by the go command, if it exists. I could see maybe only doing this for debug and test, or adding an option to override it.

Reply:
If I remember correctly, the go version used to compile the binary is stored in the binary itself, so perhaps it would be doable to read it from there instead and have it as a single check, regardless of how delve debugs the target (via exec, debug, or test)?